packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -665,7 +667,8 @@ class JavaGenerator extends Generator {
 
   protected onEndClass(cls: spec.ClassType) {
     if (cls.abstract) {
-      this.emitProxy(cls);
+      const type = this.reflectAssembly.findType(cls.fqn) as reflect.ClassType;
```

Comment:
Wow this whole file is some hairy bull@#$%.

I don't think I can even start reviewing this maze of `if` statements in any kind of meaningful sense. Guess I'll trust you and the test set. :)

Reply:
I initially tried cleaning up this mess... and tracked back after sinking almost a day into it and not being close to done. I feel you, and I am genuinely itching to re-write this whole piece entirely at this point... maybe someday...

Reply:
I get it, and I don't mind.